### PR TITLE
add debian9 as supported OS, enable acceptance tests with debian8 and debian9

### PR DESCRIPTION
**Pull request by markgraf**
added Debian 9, hope the spaceing is right :-)

#### Pull Request (PR) description
https://github.com/voxpupuli/puppet-letsencrypt/pull/170 seems to be inactive since April.
Does this help?

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -43,7 +43,8 @@
     {
       "operatingsystem": "Debian",
       "operatingsystemrelease": [
-        "8"
+        "8",
+        "9"
```
Comment (Dan33l) [inline metadata.json]: Hi. @markgraf .
Can you also enable acceptance tests for debian ?

Comment (markgraf) [reply metadata.json]: Took me a while to smarten myself up a bit :-)
Next commit should do the trick.  I have no AWS-Account for testing though. :-/

**Comment by markgraf** [comment]
`expected that the catalogue would contain Class[letsencrypt::install] with configure_epel set to true but it is set to false`

So I guess the acceptance-test needs some context?
configure_epel set to false sounds right for Debian :-)

**Comment by Dan33l** [comment]
We use `beaker-hostgenerator` to generate nodesets on the fly.
So you have to update :
 * [.travis.yml](https://github.com/voxpupuli/puppet-letsencrypt/blob/master/.travis.yml) to enable tests for the PR
* [.sync.yml](https://github.com/voxpupuli/puppet-letsencrypt/blob/master/.sync.yml) to stay enabled after the next modulesync apply.

File: .sync.yml [hunk 2]
```diff
@@ -5,5 +5,7 @@
     - set: ubuntu1404-64
     - set: ubuntu1604-64
     - set: ubuntu1804-64
+    - set: debianjessie-64
```
Comment (Dan33l) [inline .sync.yml]: should be 
```suggestion
    - set: debian8-64
```

Comment (markgraf) [reply .sync.yml]: Ninjaed :-)

File: spec/acceptance/nodesets/ec2/debian-jessie-x64.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (Dan33l) [inline spec/acceptance/nodesets/ec2/debian-jessie-x64.yml]: The file can be removed

File: spec/acceptance/nodesets/ec2/debian-stretch-x64.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (Dan33l) [inline spec/acceptance/nodesets/ec2/debian-stretch-x64.yml]: The file can be removed

File: spec/acceptance/nodesets/ec2/image_templates.yaml [hunk 5]
```diff
@@ -27,6 +27,16 @@ AMI:
     :image:
       :aio: ami-fe408091
     :region: eu-central-1
+  # Debian 8.7 (HVM),
```
Comment (Dan33l) [inline spec/acceptance/nodesets/ec2/image_templates.yaml]: This modification can be removed

Comment (markgraf) [reply spec/acceptance/nodesets/ec2/image_templates.yaml]: Line 30-34, ok.
Lines 35-39 too?

File: spec/acceptance/nodesets/ec2/debian-jessie-x64.yml (hunk 3 above)
Comment (markgraf) [reply spec/acceptance/nodesets/ec2/debian-jessie-x64.yml]: Ok, will do.

File: spec/acceptance/nodesets/ec2/debian-stretch-x64.yml (hunk 4 above)
Comment (markgraf) [reply spec/acceptance/nodesets/ec2/debian-stretch-x64.yml]: Ok, will do.

File: spec/acceptance/nodesets/ec2/image_templates.yaml (hunk 5 above)
Comment (Dan33l) [reply spec/acceptance/nodesets/ec2/image_templates.yaml]: please do not modify this file with the PR.

**Comment by Dan33l** [comment]
Accordingly with `README.md` and [https://packages.debian.org](https://packages.debian.org/search?suite=jessie&searchon=names&keywords=letsencrypt) the package was not provided with debian8.

So, with this OS only `install_method=vcs` is possible. That explain the acceptance fail for debian8. We need to update the acceptance with debian8.

**Comment by markgraf** [comment]
Debian8 can not find certbot-package.
Still failing because epel is expected. But `epel = facts[:osfamily] == 'RedHat'` and `configure_epel: epel,` should set this expectation to false, shouldn't it???

**Comment by markgraf** [comment]
> the package was not provided with debian8.
>
> So, with this OS only `install_method=vcs` is possible.

That would be in spec/classes/letsencrypt_spec.rb lines 73 and following?

**Comment by ekohl** [comment]
I also started https://github.com/voxpupuli/puppet-letsencrypt/pull/193 to clean up some things.

**Comment by ekohl** [comment]
#193 has been merged instead.

**Comment by markgraf** [comment]
Ok, I'll have to do manual installs then.  My company still uses Debian 8.  Yes, I know...

**Comment by ekohl** [comment]
You can still override the install method on Debian Jessie to set it to VCS and I'd expect that to work.

**Pull request closed without merging**